### PR TITLE
Fix doxygen compilation

### PR DESCRIPTION
**Pull request by hiergaut**
<del>All public function, member of class have to be commented (checked by doxygen).</del>
Only fix doxygen compilation for the project.
Another PR will make the full documentation.
Deploy doxygen doc to github page.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -6,6 +6,8 @@ Current head v.2.1 rc
   * Add CHANGELOG file (#14)
 - Clean code
   * Removed first server impl (#16)
+- Documentation
+  * Fixed doxygen
```
Comment (hiergaut) [inline CHANGELOG:10]:
```suggestion
  * Fixed doxygen (#20)
```

**Comment by hiergaut** [comment]
@nmellado ready to merge

**Comment by nmellado** [comment]
The added make doxygen happy, but do not add any meaningful content. 
Also, I still see methods without comments in the files.

**Comment by hiergaut** [comment]
I am going to do a full doxygen revision next days ...

**Comment by hiergaut** [comment]
waiting for #22 merged

**Comment by hiergaut** [comment]
I have some stuff to do, coming soon.

**Comment by hiergaut** [comment]
``` 
 /usr/bin/git push origin gh-pages
 remote: Permission to T-PLAY/hub.git denied to github-actions[bot].
 fatal: unable to access 'https://github.com/T-PLAY/hub.git/': The requested URL returned error: 403
 Error: Action failed with "The process '/usr/bin/git' failed with exit code 128" 
  ```
  
Unable to push to git page, can you specify me, what is the github page url please ?

**Comment by nmellado** [comment]
The action do not find the doc files
https://github.com/T-PLAY/hub/actions/runs/8567529175/job/23479489255?pr=20#step:6:873
can you have a look ?

**Pull request merged**